### ISSUE TITLE
DataPrep.Connector: Location is required when latitude and longitude are provided for Yelp

### ISSUE THREAD

**Issue by jnwang** [sfu-db/dataprep#461]
**Describe the bug**
DataPrep.Connector always requires `location` to be specified. However, according to the [Yelp API documentation](https://www.yelp.ca/developers/documentation/v3/business_search), `location`  is not required when latitude and longitude are provided. It would be better to make it consistent with the doc.

```python
from dataprep.connector import Connector

conn_yelp = Connector("./yelp", _auth={"access_token":yelp_access_token}, _concurrency = 5)

df = await conn_yelp.query("businesses", term = "resturant", latitude = "49.28617652439037", longitude = "-123.13819352127699", _count = 10)

```
<img width="819" alt="Screen Shot 2020-12-27 at 11 24 45 PM" src="https://user-images.githubusercontent.com/1028970/103197394-ca3b9d00-489a-11eb-8491-0b40192cc3f5.png">

Desktop (please complete the following information):

OS: MacOS Catalina
Browser: Chrome
Platform: Jupyter Notebook
Platform Version 6.0.1
Python Version 3.7.4
Dataprep Version 0.2.4

**Comment by dovahcrow** [comment]
Certainly, we can relax the requirement of `location` to optional, but that's not correct. Eventually, we want to have some enforcement of: either param A or param B exist.
